crates/provider/src/fillers/mod.rs

```diff
@@ -396,28 +396,19 @@ where
         self.inner.get_balance(address)
     }
 
-    async fn get_block(
-        &self,
-        block: BlockId,
-        kind: BlockTransactionsKind,
-    ) -> TransportResult<Option<N::BlockResponse>> {
-        self.inner.get_block(block, kind).await
+    fn get_block(&self, block: BlockId) -> EthGetBlock<N, Option<N::BlockResponse>> {
```

Comment:
why does this need an N generic and an additional generic for N::BlockResponse, doesn't make much sense